### PR TITLE
Main Commit for Colossus Component

### PR DESCRIPTION
**Pull request by malibuworkcrew**
* CORS support
* Encoders
* Remote address header
* Easy Response wrapper
* Response serialization

@ladinu

### CODE REVIEW

File: src/main/scala/com/webtrends/harness/component/colossus/http/CORS.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                           allowedMethods: Seq[HttpMethod] = Seq.empty,
+                           accessControlAllowHeaders: AllowedHeaders = AllowedHeaders.`*`)
+
+trait CORS extends HttpBodyEncoders {
+  def cors(s: Settings, inner: PartialFunction[HttpRequest, Future[ColossusResponse]])
+  : PartialFunction[HttpRequest, Future[ColossusResponse]] = {
+    case req =>
+      (req.head.method, req.head.headers.firstValue("Origin"), req.head.headers.firstValue("Access-Control-Request-Method")) match {
+        case (HttpMethod.Options, Some(origin), Some(requestMethod)) => // Case 1: pre-flight request
+          Future {
```
Comment (ladinu) [inline src/main/scala/com/webtrends/harness/component/colossus/http/CORS.scala]: Should probably be `Future.successful(...)`

File: src/main/scala/com/webtrends/harness/component/colossus/http/RemoteAddress.scala [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import colossus.protocols.http.HttpHeader
+import com.webtrends.harness.command.Command
+import com.webtrends.harness.component.colossus.command.{ColossusCommand, ColossusResponse}
+
+trait RemoteAddress extends ColossusCommand { this: Command =>
+  override def postProcessing(resp: ColossusResponse) = {
+    super.postProcessing(resp)
+    resp.copy(headers = resp.headers +
+      HttpHeader("Remote-Address", InetAddress.getLocalHost.getHostAddress))(resp.formats)
```
Comment (ladinu) [inline src/main/scala/com/webtrends/harness/component/colossus/http/RemoteAddress.scala]: Not sure why we want `Remote-Address` appended to each http response. We do need `Remote-Address` in the http request though if present.

Comment (ladinu) [inline src/main/scala/com/webtrends/harness/component/colossus/http/RemoteAddress.scala]: You can do something like this to get the client address 

```scala
connection.connectionState match {
    case state: AliveState => state
      .endpoint
      .remoteAddress
      .map(_.getHostString)
      .map(hostString => req.withHeader("Remote-Address", hostString))
   case _ => None
```

**Comment by ladinu** [comment]
Nice job!

**Comment by malibuworkcrew** [comment]
@ladinu Thanks for the review! Think I got both your comments in 'Remote address put on request instead of response' commit

**Comment by ladinu** [comment]
@malibuworkcrew Looks good. Unfortunately I don't have merge rights.

**Pull request merged**